### ISSUE TITLE
C interop few C macro skipped by cinterop

### ISSUE THREAD

**Issue by dickensas** [JetBrains/kotlin-native#3608]
I tried to interop Lua C API  with the below Def file

headers = luaconf.h lauxlib.h lualib.h
package = lua
linkerOpts = -llua
compilerOpts = -llua 

The macro define in lauxlib.h

#define luaL_dofile(L, fn) \\
	(luaL_loadfile(L, fn) || lua_pcall(L, 0, LUA_MULTRET, 0))

cinterop skips the above macro leads to missing luaL_dofile definition
also leads to almost all those methods missing

Kotlin Version I am using: 1.3.70-dev-1231-17

**Comment by dickensas** [comment]
I have downloaded Lua 5.1 library using pacman -S mingw-w64-x86_64-lua51

and used those header files, now the definitions are available, somehow those definitions are not interoped for lua for 5.3

**Comment by msink** [comment]
```
---
#undef luaL_dofile
static inline int luaL_dofile(lua_State *L, const char *filename) {
        return (luaL_loadfile(L, fn) || lua_pcall(L, 0, LUA_MULTRET, 0));
}
```

**Comment by dickensas** [comment]
it is working but a simple Lua -print("Hello World!!")
is no printing for the below code, no errors printed
any flags to make it visible ?
below is the kotlin code

import lua.* //def file interop
import kotlinx.cinterop.*

fun main() {
    var L = luaL_newstate()
    luaL_openlibs(L)
    luaL_loadfile(L, "fact.lua")
    lua_close(L)
}

**Comment by dickensas** [comment]
It is fixed, for Lua 5.1 I need to call lua_pcall(L, 0, LUA_MULTRET, 0)
for Lua 5.3 the solution provided by @msink is working

**Comment by dickensas** [comment]
The final code for Lua 5.3 def file is
---
#undef luaL_dofile
static inline int luaL_dofile(void *L, const char *fn) {
    return luaL_loadfilex(L, fn, NULL) || lua_pcallk(L, 0, LUA_MULTRET, 0, 0, NULL);
}

new function names in Lua 5.3 are
luaL_loadfilex
lua_pcallk
